core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java

```diff
@@ -15,204 +15,294 @@
+        {
+            this.pageCount = pageCount;
+        }
 
-        private int getPageCount()
+        public void setType(TestType type)
         {
-            return PAGE_COUNT;
+            this.type = type;
         }
 
         public Page getDataPage()
         {
             return dataPage;
         }
 
-        private PartitionedOutputOperator createPartitionedOutputOperator(OperatorFactories operatorFactories)
+        public enum TestType
+        {
+            BIGINT(BigintType.BIGINT, 5000),
+            DICTIONARY_BIGINT(BigintType.BIGINT, 3000) {
+                @Override
+                public Page createPage(List<Type> types, int positionCount, float nullRate)
+                {
+                    return createRandomDictionaryPage(types, POSITION_COUNT, nullRate);
+                }
+            },
+            RLE_BIGINT(BigintType.BIGINT, 3000) {
+                @Override
+                public Page createPage(List<Type> types, int positionCount, float nullRate)
+                {
+                    return createRandomRlePage(types, POSITION_COUNT, nullRate);
+                }
+            },
+            LONG_DECIMAL(createDecimalType(MAX_SHORT_PRECISION + 1), 5000),
+            INTEGER(IntegerType.INTEGER, 5000),
+            SMALLINT(SmallintType.SMALLINT, 5000),
+            BOOLEAN(BooleanType.BOOLEAN, 5000),
+            VARCHAR(VarcharType.VARCHAR, 5000),
+            ARRAY_BIGINT(new ArrayType(BigintType.BIGINT), 1000),
+            ARRAY_VARCHAR(new ArrayType(VarcharType.VARCHAR), 1000),
+            ARRAY_ARRAY_BIGINT(new ArrayType(new ArrayType(BigintType.BIGINT)), 1000),
+            MAP_BIGINT_BIGINT(createMapType(BigintType.BIGINT, BigintType.BIGINT), 1000),
+            MAP_BIGINT_MAP_BIGINT_BIGINT(createMapType(BigintType.BIGINT, createMapType(BigintType.BIGINT, BigintType.BIGINT)), 1000),
+            ROW_BIGINT_BIGINT(rowTypeWithDefaultFieldNames(ImmutableList.of(BigintType.BIGINT, BigintType.BIGINT)), 1000),
+            ROW_ARRAY_BIGINT_ARRAY_BIGINT(rowTypeWithDefaultFieldNames(ImmutableList.of(new ArrayType(BigintType.BIGINT), new ArrayType(BigintType.BIGINT))), 1000);
+
+            private final Type type;
+            private final int pageCount;
+
+            TestType(Type type, int pageCount)
+            {
+                this.type = requireNonNull(type, "type is null");
+                this.pageCount = pageCount;
+            }
+
+            public Page createPage(List<Type> types, int positionCount, float nullRate)
+            {
+                return createRandomPage(types, POSITION_COUNT, nullRate);
+            }
+
+            public Type getType()
+            {
+                return type;
+            }
+
+            public int getPageCount()
+            {
+                return pageCount;
+            }
+        }
+
+        @Setup
+        public void setup(Blackhole blackhole) // blackhole has to be injected by jmh (should not be create3d manually)
```

Comment:
`create3d` -> `created`

Move the comment above:
```
this.blackhole = blackhole;
```
also add:
```
We don't check blackhole is not null, because...
```